discovery_engine_core/core/src/stack/ops/personalized.rs

```diff
@@ -66,18 +67,41 @@ impl Ops for PersonalizedNews {
                 filter.add_keyword(kp.words())
             });
 
+            let mut results = FuturesUnordered::new();
+
             for market in markets.read().await.iter() {
-                let query = NewsQuery {
-                    market,
-                    filter,
-                    page_size: self.page_size,
-                    page: None,
-                };
-                match self.client.news(&query).await {
-                    Ok(batch) => articles.extend(batch),
-                    Err(err) => errors.push(err),
+                // a lot of cloning
+                // if NewsQuery can own the market + filter we can create the
```

Comment:
I'm not sure having `NewsQuery` taking ownership will solve the problem. We still need to clone the market and the filter in that. 

The `Client` is stateless so we can put it in an `Arc` and avoid to clone it. But for the rest is not a problem we have max 3 markets.

Reply:
We can change `NewsQuery` to take `Deref<Target = Filter>` instead so we can avoid to clone it but seems more a micro optimization at the moment.

Reply:
> I'm not sure having NewsQuery taking ownership will solve the problem. We still need to clone the market and the filter in that.

Yes, we unfortunately need to clone markets and filters because tokio::spawn requires the future to be `'static` but if NewsQuery has the ownership of `market`, `filter` we can save some lines like 
```rust
let market_inner = market;
let filter_inner = filter;
```

What do you mean by `Deref<Target = Filter>`?

I will put the `Client` behind an `Arc` and move the creation of the request in a function. that should clean up some stuff.

Reply:
> Yes, we unfortunately need to clone markets and filters because tokio::spawn requires the future to be `'static` but if NewsQuery has the ownership of `market`, `filter` we can save some lines like
> 
> ```rust
> let market_inner = market;
> let filter_inner = filter;
> ```

Maybe we can save some line but the clone will be there anyway no?

> What do you mean by `Deref<Target = Filter>`?

If we have something like
```rust
NewsQuery<'a, F> {
  market: &'a,
  filter: F
}

async fn headlines(q: NewsQuery<'_, impl Deref<Target = Filter>) { ... }
```

Then we can put the `Filter` in an `Arc` and construct the `NewsQuery` with that.

Reply:
implemented it for the filter